### DISCUSSION TITLE
@Recurring with application.properties?

### DISCUSSION THREAD

**Discussion by VN11733** [jobrunr/jobrunr#1141] [Q&A] [answered]
Hi community,
we build an app with jobrunr-pro (7.2.3) and quarkus (3.8.4) on board and we'd like to configure scheduled jobs with `@Recurring` annotation, but we're not able to find a way how to provide cron from `applicaiton.properties` to annotation.
Is it possible to do so? If not, I'd like to request such feature (if possible) since it would be handy to place job crons and ids inside `application.properites`.

Thank you.

**Comment by auloin** [comment]
I know this is possible in Spring (see example https://github.com/jobrunr/example-order-fulfillment/blob/main/src/main/java/org/jobrunr/example/tasks/OrderFulfillmentTasks.java#L54). We haven't tried it for Quarkus but it should similarly work (see https://quarkus.io/guides/scheduler-reference#intervals)?

**Reply by VN11733** [reply]
> I know this is possible in Spring (see example https://github.com/jobrunr/example-order-fulfillment/blob/main/src/main/java/org/jobrunr/example/tasks/OrderFulfillmentTasks.java#L54). We haven't tried it for Quarkus but it should similarly work (see https://quarkus.io/guides/scheduler-reference#intervals)?

Unfortunately, Quarkus doesn't support placeholders...

**Reply by rdehuyss** [reply]
> Unfortunately, Quarkus doesn't support placeholders...

Hmm, I notice we have [unit tests](https://github.com/jobrunr/jobrunr/blob/master/framework-support/jobrunr-quarkus-extension/quarkus-jobrunr/src/test/java/org/jobrunr/scheduling/JobRunrRecurringJobRecorderTest.java#L150-L186) for this though. This worked back in 2021 but I know Quarkus changes quite some internals.

Can you create a repo that reproduces your issue?

**Reply by VN11733** [reply]
> > Unfortunately, Quarkus doesn't support placeholders...
> 
> Hmm, I notice we have [unit tests](https://github.com/jobrunr/jobrunr/blob/master/framework-support/jobrunr-quarkus-extension/quarkus-jobrunr/src/test/java/org/jobrunr/scheduling/JobRunrRecurringJobRecorderTest.java#L150-L186) for this though. This worked back in 2021 but I know Quarkus changes quite some internals.
> 
> Can you create a repo that reproduces your issue?

I'll try my best to do it in a few days. I'll let you know

P.S. your test only covers `jobRunrRecurringJobRecorder::schedule` method whereas my problem is with `@Recurring` annotation.

**Reply by rdehuyss** [reply]
> P.S. your test only covers jobRunrRecurringJobRecorder::schedule method whereas my problem is with @Recurring annotation.
These are the same thing. Testing Quarkus Extensions is quite difficult.

**Reply by VN11733** [reply]
Ok, I've double checked and **it is possible to set cron as placeholder**, however, it is not for id:

```log
Caused by: java.lang.IllegalArgumentException: The id of a recurring job can only contain letters and numbers.
	at org.jobrunr.jobs.RecurringJob.validateAndSetId(RecurringJob.java:300)
	at org.jobrunr.jobs.RecurringJob.<init>(RecurringJob.java:99)
	at org.jobrunr.jobs.RecurringJob.<init>(RecurringJob.java:83)
	at org.jobrunr.scheduling.JobRunrRecurringJobRecorder.schedule(JobRunrRecurringJobRecorder.java:46)
	at io.quarkus.deployment.steps.JobRunrExtensionProcessor$findRecurringJobAnnotationsAndScheduleThem213566530.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.JobRunrExtensionProcessor$findRecurringJobAnnotationsAndScheduleThem213566530.deploy(Unknown Source)
	... 13 more
```

**Reply by rdehuyss** [reply, marked as answer]
Which makes sense from our point of view...

**Reply by VN11733** [reply]
So it is not a bug, but feature. Got it, thanks! Marking it as an answer :)
